patterns/grok-patterns

```diff
@@ -48,7 +48,7 @@ URI %{URIPROTO}://(?:%{USER}(?::[^@]*)?@)?(?:%{URIHOST})?(?:%{URIPATHPARAM})?
 MONTH \b(?:Jan(?:uary)?|Feb(?:ruary)?|Mar(?:ch)?|Apr(?:il)?|May|Jun(?:e)?|Jul(?:y)?|Aug(?:ust)?|Sep(?:tember)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?)\b
 MONTHNUM (?:0?[1-9]|1[0-2])
 MONTHNUM2 (?:0[1-9]|1[0-2])
-MONTHDAY (?:(?:0[1-9])|(?:[12][0-9])|(?:3[01])|[1-9])
+MONTHDAY (?:(?:(0|\s)?[1-9])|(?:[12][0-9])|(?:3[01])|[1-9])
```

Comment:
Also, I agree that the whitespace match probably should not be part of the MONTHDAY match.

On a side note, I think that the original:

    (?:(?:0[1-9])|(?:[12][0-9])|(?:3[01])|[1-9])

is equivalent to the following:

    (?:(?:0?[1-9])|(?:[12][0-9])|(?:3[01]))

I don't know if the shorter one is an improvement though.